controllers/kubeadmconfig_controller.go

```diff
@@ -119,43 +120,7 @@ func (r *KubeadmConfigReconciler) Reconcile(req ctrl.Request) (ctrl.Result, erro
 		return ctrl.Result{}, err
 	}
 
-	// Store Config's state, pre-modifications, to allow patching
-	patchConfig := client.MergeFrom(config.DeepCopy())
-
-	// if the machine has cluster and or init defined on it then generate the init regular join
-	if config.Spec.InitConfiguration != nil && config.Spec.ClusterConfiguration != nil {
-		// get both of these to strings to pass to the cloud init control plane generator
-		initdata, err := json.Marshal(config.Spec.InitConfiguration)
-		if err != nil {
-			log.Error(err, "failed to marshal init configuration")
-			return ctrl.Result{}, err
-		}
-		clusterdata, err := json.Marshal(config.Spec.ClusterConfiguration)
-		if err != nil {
-			log.Error(err, "failed to marshal cluster configuration")
-			return ctrl.Result{}, err
-		}
-
-		cloudInitData, err := cloudinit.NewInitControlPlane(&cloudinit.ControlPlaneInput{
-			InitConfiguration:    initdata,
-			ClusterConfiguration: clusterdata,
-		})
-		if err != nil {
-			log.Error(err, "failed to generate cloud init for bootstrap control plane")
-			return ctrl.Result{}, err
-		}
-
-		config.Status.BootstrapData = cloudInitData
-		config.Status.Ready = true
-
-		if err := r.Update(ctx, &config); err != nil {
-			log.Error(err, "failed to update config")
-			return ctrl.Result{}, err
-		}
-		return ctrl.Result{}, nil
-	}
-
-	// Check for control-plane ready. If it's not ready then we will requeue the machine until it is.
+	// Check for infrastructure ready. If it's not ready then we will requeue the machine until it is.
 	// The infrastructure provider *must* set this value to use this bootstrap provider.
 	if cluster.Annotations == nil {
```

Comment:
We maybe can get rid of this check entirely. I'd say it's safe to generate the bootstrap data regardless of the cluster infrastructure's status. We can move this burden to each infrastructure provider - they should not proceed to create instances from Machines until the Cluster infra is ready. WDYT?

Reply:
I'm assuming we should wait for the load balancer to get created in order to use it for `ClusterConfiguration.ControlPlaneEndpoint`

Reply:
I just opened #85 but you raise a good point - let's continue the conversation there.

Reply:
Nevermind - we decided to keep the cluster status infraReady check